### ISSUE TITLE
Release into Indigo?

### ISSUE THREAD

**Issue by rctoris** [ipa320/softkinetic#40]
We have been using the package on Indigo with Ubuntu 14.04 and we'd like to depend on it with some of our release software. Are there any plans to release this into the build farm soon?

**Comment by floweisshardt** [comment]
@corot: I can help releasing the repository and get it through the buildfarm. All I need you to do is to push the version you would like to release into the indigo_release_candidate branch (Just open a PR and I'll review and accept according to release aspects).

**Comment by corot** [comment]
I'm :+1: for release indigo, but I think we have a problem with the softkinetic driver: afaik, it cannot be installed with apt-get, just by hand, so jenkins will always fail to build it. Hopefully I'm wrong and there's a way!

Btw, @rctoris are you using hydro or indigo branch? They're quite different now.

**Comment by gavanderhoorn** [comment]
@corot: `ueye_cam` faced similar difficulties. See anqixu/ueye_cam#10, and anqixu/ueye_cam@97cfc9d7 for how they (he) fixed it.

I'm not sure whether this is actually 'best practice', but it is at least something workable.

**Comment by corot** [comment]
First word to come to my mind is :  "intimidating". Doesn't look like an easy solution, and I'm too short of time now to try it. But let's keep this issue open and hopefully me or someone else will find time to try!

Another good reason for waiting is to see whether they succeed! :grin:

**Comment by gavanderhoorn** [comment]
Well it's just a script that downloads the required files and extracts them. I'm not sure anymore whether SoftKinetic have their downloads behind some registration / login system. In that case, it wouldn't work.

> Another good reason for waiting is to see whether they succeed!

That script is actually a copy of the one used in the `pointgrey` driver, so it has already been used.

Just documenting here.

**Comment by corot** [comment]
Yes, still needs logging/password for Linux drivers, but not for Windows ones. We could ask them to remove this restriction.

**Comment by rctoris** [comment]
@corot we are using the indigo_dev branch.

**Comment by corot** [comment]
Hi, I saw [this repo](https://github.com/ph4m/DepthSenseGrabber). They don't install automatically the DepthSense SDK but at least confirm that it is installed. We can easily do the same.

**Comment by gavanderhoorn** [comment]
@corot: isn't that already done in [CMakeLists.txt - L8-L16](https://github.com/ipa320/softkinetic/blob/484eb9e782a82cba9c1c451be241be4c7ec4e5bd/softkinetic_camera/CMakeLists.txt#L8-L16)?

**Comment by corot** [comment]
Right! My bad. Is it ok to release like this? Or we break any ROS convention not providing automatic installing of deps?

**Comment by gavanderhoorn** [comment]
The issue is (iirc) that you cannot build a deb of this pkg on the buildfarm as the softkinetic sdk is missing there. The build will error out with the message I linked earlier and no deb will be produced.

I'm not sure what the best course of action is. I remember suggesting linking the Softkinetic SDK dynamically (so don't link at build time), which would remove the need for the `.so`'s to be present on the buildfarm (typedefs etc would probably need to be duplicated), and I also remember some discussion around that but I can't find it anymore.

**Comment by corot** [comment]
Yes... I should have re-read the previous posts before bothering anyone, sorry.
Right, this is a blocking point. The solution seems to clone the trick of the people @gavanderhoorn mentioned on the post of Oct 16, 2014, assuming we can download the SDK in the same way.

This can be a good amount of work, plus regular updates for new versions. Do you think is worth the effort and time? I don't know how many people is out there using the Senz3D with ROS, nor if more people would join in case of having release packages. If so, I will at least test the automatic download and installation to ensure it is feasible. If not, there are other, maybe more interesting improvements, as dynamic reconfigure and move filters to a nodelet.

**Comment by gavanderhoorn** [comment]
I can't comment on the nr of people using the driver. Seeing as this is a source-only pkg, you could perhaps use the [graphs/traffic](https://github.com/ipa320/softkinetic/graphs/traffic) page for some statistics that might provide some insight.

Downloading the sdk on build seems like a viable option, provided it is actually possible with the Sense3D sdk.
